x/leverage/keeper/keeper.go

```diff
@@ -265,32 +266,29 @@ func (k Keeper) BorrowAsset(ctx sdk.Context, borrowerAddr sdk.AccAddress, borrow
 	return nil
 }
 
-// RepayAsset attempts to repay an open borrow position with base assets. If asset type is invalid,
-// account balance is insufficient, or no open borrow position exists, we return an error.
+// RepayAsset attempts to repay an open borrow position. Payment denom must be one of the
+// borrowed asset. If asset type is invalid, account balance is insufficient, or no open
+// borrow position exists, we return an error.
 // Additionally, if the amount provided is greater than the full repayment amount, only the
 // necessary amount is transferred. Because amount repaid may be less than the repayment attempted,
 // RepayAsset returns the actual amount repaid.
 func (k Keeper) RepayAsset(ctx sdk.Context, borrowerAddr sdk.AccAddress, payment sdk.Coin) (sdk.Int, error) {
 	if !payment.IsValid() {
-		return sdk.ZeroInt(), sdkerrors.Wrap(types.ErrInvalidAsset, payment.String())
-	}
-
-	if !k.IsAcceptedToken(ctx, payment.Denom) {
-		return sdk.ZeroInt(), sdkerrors.Wrap(types.ErrInvalidAsset, payment.String())
```

Comment:
this check can be a source of bugs. We should always be able to close pay off a borrow position.

Reply:
This has also been added to
https://github.com/umee-network/umee/pull/913